datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -2130,16 +2130,10 @@ impl Filter {
             }
         }
 
-        // filter predicates should not be aliased
-        if let Expr::Alias(Alias { expr, name, .. }) = predicate {
-            return plan_err!(
-                "Attempted to create Filter predicate with \
-                expression `{expr}` aliased as '{name}'. Filter predicates should not be \
-                aliased."
-            );
-        }
-
-        Ok(Self { predicate, input })
+        Ok(Self {
+            predicate: predicate.unalias_nested().data,
```

Comment:
My thought is to execute `unalias_nested` when creating the initial logical plan. 

But the current implementation is also acceptable to me. Thank you @samuelcolvin .  If we follow the current implementation, we also need to remove the redundant `unalias_nested` calls outside of try_new.

cc @alamb

Reply:
Filed https://github.com/apache/datafusion/issues/11339 to track

Reply:
PR to fix: https://github.com/apache/datafusion/pull/11340